tests/tailor.js

```diff
@@ -687,4 +691,68 @@ describe('Tailor', () => {
         });
     });
 
+    it('should unzip the fragment response if it is compressed', (done) => {
+        nock('https://fragment')
+            .get('/1').reply(200, 'hello')
+            .defaultReplyHeaders({
+                'content-encoding': 'gzip',
+            })
+            .get('/2')
+            .reply(200, ()=> {
+                return zlib.gzipSync('GZIPPED');
+            });
+
+        mockTemplate
+            .returns(
+                '<fragment src="https://fragment/1"></fragment>' +
+                '<fragment src="https://fragment/2"></fragment>'
+            );
+
+        getResponse('http://localhost:8080/test').then((response) => {
+            assert.equal(response.body,
+                '<html>' +
+                '<head></head>' +
+                '<body>' +
+                '<script data-pipe>p.start(0)</script>' +
+                'hello' +
+                '<script data-pipe>p.end(0)</script>' +
+                '<script data-pipe>p.start(1)</script>' +
+                'GZIPPED' +
+                '<script data-pipe>p.end(1)</script>' +
+                '</body>' +
+                '</html>'
+            );
+            done();
+        });
+    });
+
+    it('should close the streams properly during unzping error', (done) => {
+        nock('https://fragment')
+            .defaultReplyHeaders({
+                'content-encoding': 'gzip',
+            })
+            .get('/2')
+            .reply(200, ()=> {
+                return new Error('GZIP Error');
```

Comment:
oh, really, you totally right. interesting how nock provides error object as a response..